ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/common/trielog/TrieLogPruner.java

```diff
@@ -83,38 +90,82 @@ public TrieLogPruner(
+            LOG.atWarn()
+                .setMessage(
+                    "Timeout occurred while loading and processing {} trie logs from database")
+                .addArgument(loadingLimit)
+                .log();
+          };
+
+      final ScheduledFuture<?> timeoutFuture =
+          preloadExecutor.schedule(timeoutTask, PRELOAD_TIMEOUT_IN_SECONDS, TimeUnit.SECONDS);
+      LOG.atInfo()
+          .setMessage(
+              "Trie log pruning will timeout after {} seconds. If this is timing out, consider using `besu storage trie-log prune` subcommand, see https://besu.hyperledger.org/public-networks/how-to/bonsai-limit-trie-logs")
+          .addArgument(PRELOAD_TIMEOUT_IN_SECONDS)
+          .log();
+
+      preloadQueue(timeoutOccurred, timeoutFuture);
+    }
+  }
+
+  private void preloadQueue(
+      final AtomicBoolean timeoutOccurred, final ScheduledFuture<?> timeoutFuture) {
+
     try (final Stream<byte[]> trieLogKeys = rootWorldStateStorage.streamTrieLogKeys(loadingLimit)) {
-      final AtomicLong count = new AtomicLong();
+
+      final AtomicLong addToPruneQueueCount = new AtomicLong();
       final AtomicLong orphansPruned = new AtomicLong();
       trieLogKeys.forEach(
           blockHashAsBytes -> {
+            if (timeoutOccurred.get()) {
+              throw new RuntimeException(
+                  new TimeoutException("Timeout occurred while preloading trie log prune queue"));
+            }
             final Hash blockHash = Hash.wrap(Bytes32.wrap(blockHashAsBytes));
             final Optional<BlockHeader> header = blockchain.getBlockHeader(blockHash);
             if (header.isPresent()) {
               addToPruneQueue(header.get().getNumber(), blockHash);
-              count.getAndIncrement();
+              addToPruneQueueCount.getAndIncrement();
             } else {
               // prune orphaned blocks (sometimes created during block production)
               rootWorldStateStorage.pruneTrieLog(blockHash);
               orphansPruned.getAndIncrement();
               prunedOrphanCounter.inc();
             }
           });
+
+      timeoutFuture.cancel(true);
       LOG.atDebug().log("Pruned {} orphaned trie logs from database...", orphansPruned.intValue());
-      LOG.atInfo().log("Loaded {} trie logs from database", count);
-      return pruneFromQueue() + orphansPruned.intValue();
+      LOG.atInfo().log(
+          "Added {} trie logs to prune queue. Commencing pruning of eligible trie logs...",
+          addToPruneQueueCount.intValue());
+      int prunedCount = pruneFromQueue();
+      LOG.atInfo().log("Pruned {} trie logs.", prunedCount);
     } catch (Exception e) {
-      LOG.error("Error loading trie logs from database, nothing pruned", e);
-      return 0;
+      if (e.getCause() != null && e.getCause() instanceof TimeoutException) {
+        int prunedCount = pruneFromQueue();
+        LOG.atInfo().log("Operation timed out, but still pruned {} trie logs.", prunedCount);
```

Comment:
```suggestion
        LOG.atInfo().log("Pruned {} trie logs before stopping due to time limit.", prunedCount);
```

Reply:
I've used the phrase "time out" in several logs (shown in description) so would like to keep it consistent

Reply:
Logs without the metadata...

```
"TrieLogPruner","message":"Attempting to load first 5000 trie logs from database..."
"TrieLogPruner","message":"Trie log pruning will timeout after 30 seconds. If this is timing out, consider using `besu storage trie-log prune` subcommand, see https://besu.hyperledger.org/public-networks/how-to/bonsai-limit-trie-logs"
"TrieLogPruner","message":"Timeout occurred while loading and processing 5000 trie logs from database","throwable"
"TrieLogPruner","message":"Operation timed out, but still pruned 513 trie logs."
```